File: src/main/java/fr/tikione/c2e/model/web/Magazine.java
package fr.tikione.c2e.model.web;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

/**
 * CanardPC web.
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Magazine {
    
    private int number;
    private String title;
    private String login;
    
    private Edito edito;
    private List<TocCategory> toc = new ArrayList<>();
}

File: src/main/kotlin/fr/tikione/c2e/model/web/Magazine.kt
package fr.tikione.c2e.model.web

/**
 * CanardPC web.
 */
class Magazine {

    var number: String = ""
    var title: String? = null
    var login: String? = null
    var edito: Edito? = null
    var toc = ArrayList<TocCategory>()

    constructor() {
    }

    constructor(number: String, title: String?, login: String?, edito: Edito?, toc: ArrayList<TocCategory>) {
        this.number = number
        this.title = title
        this.login = login
        this.edito = edito
        this.toc = toc
    }

    override fun toString(): String {
        return "Magazine(number=$number, title=$title, login=$login, edito=$edito, toc=$toc)"
    }
}
